join-domain/elx/pbis/files/join.sh

```diff
@@ -21,116 +21,275 @@
+then
+  EXCLUSIVEARGS=TRUE
+  UsageMsg
+fi
+
+eval set -- "${OPTIONBUFR}"
+
+###################################
+# Parse contents of ${OPTIONBUFR}
+###################################
+while true
+do
+  case "$1" in
+      -h|--help)
+        UsageMsg
+        ;;
+      -n|--netbios-name)
+        case "$2" in
+            "")
+              logIt "Error: option required but not specified" 1
+              shift 2;
+              exit 1
+              ;;
+            *)
+              DOMSHORT="${2}"
+              shift 2;
+              ;;
+        esac
+        ;;
+      -f|--hostname)
+        case "$2" in
+            "")
+              logIt "Error: option required but not specified" 1
+              shift 2;
+              exit 1
+              ;;
+            *)
+              DOMFQDN="${2}"
+              shift 2;
+              ;;
+        esac
+        ;;
+      -u|--join-user)
+        case "$2" in
+            "")
+              logIt "Error: option required but not specified" 1
+              shift 2;
+              exit 1
+              ;;
+            *)
+              SVCACCT="${2}"
+              shift 2;
+              ;;
+        esac
+        ;;
+      -c|--join-crypt)
+        case "$2" in
+            "")
+              logIt "Error: option required but not specified" 1
+              shift 2;
+              exit 1
+              ;;
+            *)
+              PWCRYPT="${2}"
+              SVCPASS="TOBESET"
+              shift 2;
+              ;;
+        esac
+        ;;
+      -k|--join-key)
+        case "$2" in
+            "")
+              logIt "Error: option required but not specified" 1
+              shift 2;
+              exit 1
+              ;;
+            *)
+              PWUNLOCK="${2}"
+              shift 2;
+              ;;
+        esac
+        ;;
+      -p|--join-password)
+        case "$2" in
+            "")
+              logIt "Error: option required but not specified" 1
+              shift 2;
+              exit 1
+              ;;
+            *)
+              SVCPASS="${2}"
+              shift 2;
+              ;;
+        esac
+        ;;
+      -o|--ou-path)
+        case "$2" in
+            "")
+              JOINOU="UNDEF"
+              shift 2;
+              ;;
+            *)
+              JOINOU="${2}"
+              JOINOU=${JOINOU// /\ }
+              shift 2;
+              ;;
+        esac
+        ;;
+      --)
+        shift
+        break
+        ;;
+      *)
+        logIt "Missing value" 1
+        exit 1
+        ;;
+  esac
+done
+
+# Check that mandatory options have been passed
+if [[ -z  ${DOMSHORT+x} ]] ||
+  [[ -z  ${DOMFQDN+x} ]] ||
+  [[ -z  ${SVCACCT+x} ]]
+then
+  MISSINGARGS=true
+  UsageMsg
+fi
+
+# Decrypt our query password (as necessary)
+if [[ ${SVCPASS} == TOBESET ]]
+then
+  SVCPASS="$(PWdecrypt)"
+  export SVCPASS
+
+  # Bail if needed decrypt failed
+  if [[ ${SVCPASS} == FAILURE ]]
```

Comment:
`PWdecrypt` does not return FAILURE on error, it returns an empty string. The prior test was functioning...
```suggestion
  if [[ -z "${SVCPASS}" ]]
```

Reply:
Copy pasta error on my part. Fixed.